v1.7.1 release checklist

## Release Checklist

[Backlog Status](https://github.com/orgs/kubearmor/projects/34/views/1)

- [ ] Manual Tests
    - [x] EKS - BottleRocket
    - [x] EKS - Gravitron (ARM64)
    - [x] EKS - Amazon Linux 2
    - [x] GKE COS (AppArmor)
    - [x] GKE COS (BPF-LSM)
    - [x] RHEL (certain BPF_LSM primitives are not available on RHEL)
    - [ ] minikube - VM Based 
    - [x] AKS Cluster - Ubuntu
    - [x] AKS Cluster - Azure Linux - No AppArmor packages
    - [ ] VM support - test kubearmor packages without k8s on RHEL and Debian based systems
    - [ ] VM Support - Docker Compose
    - [ ] KubeArmor Performance [Benchmarking Data](https://github.com/kubearmor/KubeArmor/wiki/KubeArmor-Performance-Benchmarking-Data) for BPF-LSM
- [ ] MarketPlace Image updates
    - [ ] [AWS/EKS Marketplace](https://aws.amazon.com/marketplace/pp/prodview-5npox4ixz64qk)
    - [ ] [RedHat Catalog](https://catalog.redhat.com/software/container-stacks/detail/6368860f558779d5f81e157e)
    - [ ] [MicroK8s Plugin](https://github.com/canonical/microk8s-community-addons/blob/main/addons/kubearmor/)
    - [ ] [Digital Ocean 1ClickApp](https://marketplace.digitalocean.com/apps/kubearmor)
- [ ] [Mark as stable release](https://github.com/kubearmor/KubeArmor/blob/main/STABLE-RELEASE)
- [ ] Update to getting-started guide for helm
- [ ] Does it require a manual update to [Operator bundle](https://github.com/kubearmor/KubeArmor/tree/main/pkg/KubeArmorOperator/bundle)? (changes required)
- [ ] Check helm charts has been released

> Note: Release checklist is needed since testing of certain platforms is not automated in CI env due to non-technical (primarily cost) concerns.

Environment: AKS
Orchestration system: Orchestrated
Tests: https://github.com/kubearmor/KubeArmor/tree/main/tests/k8s_env
```
kubectl get nodes -o wide

NAME                             STATUS   ROLES    AGE   VERSION   INTERNAL-IP   EXTERNAL-IP   OS-IMAGE                    KERNEL-VERSION      CONTAINER-RUNTIME
aks-azure-33470057-vmss000000    Ready    <none>   63m   v1.34.7   10.224.0.5    <none>        Microsoft Azure Linux 3.0   6.6.130.1-3.azl3    containerd://2.0.0
aks-ubuntu-33470057-vmss000000   Ready    <none>   63m   v1.34.7   10.224.0.4    <none>        Ubuntu 22.04.5 LTS          5.15.0-1110-azure   containerd://1.7.31-1
```

<details>
<summary>karmor probe</summary>

```
FFound KubeArmor running in Kubernetes

Daemonset :
 	kubearmor 	Desired: 1	Ready: 1	Available: 1	
Deployments : 
 	kubearmor-controller	Desired: 1	Ready: 1	Available: 1	
 	kubearmor-operator  	Desired: 1	Ready: 1	Available: 1	
 	kubearmor-relay     	Desired: 1	Ready: 1	Available: 1	
Containers : 
 	kubearmor-bpf-containerd-98c2c-xt5nv     	Running: 1	Image Version: kubearmor/kubearmor:v1.7.1                       	
 	kubearmor-controller-676669fd5b-w5xzz    	Running: 1	Image Version: docker.io/kubearmor/kubearmor-controller:v1.7.1  	
 	kubearmor-operator-7d9745dcbc-5mdm7      	Running: 1	Image Version: docker.io/kubearmor/kubearmor-operator:v1.7.1    	
 	kubearmor-relay-6f5445f967-z8nxk         	Running: 1	Image Version: docker.io/kubearmor/kubearmor-relay-server:latest	
 	kubearmor-apparmor-containerd-98c2c-9z4w2	Running: 1	Image Version: kubearmor/kubearmor:v1.7.1                       	
Node 1 : 
 	OS Image:                 	Ubuntu 22.04.5 LTS   	
 	Kernel Version:           	5.15.0-1110-azure    	
 	Kubelet Version:          	v1.34.7              	
 	Container Runtime:        	containerd://1.7.31-1	
 	Active LSM:               	AppArmor             	
 	Host Security:            	false                	
 	Container Security:       	true                 	
 	Container Default Posture:	block(File)          	block(Capabilities)	block(Network)	
 	Host Default Posture:     	audit(File)          	audit(Capabilities)	audit(Network)	
 	Host Visibility:          	none                 	
Node 2 : 
 	OS Image:                 	Microsoft Azure Linux 3.0	
 	Kernel Version:           	6.6.130.1-3.azl3         	
 	Kubelet Version:          	v1.34.7                  	
 	Container Runtime:        	containerd://2.0.0       	
 	Active LSM:               	BPFLSM                   	
 	Host Security:            	false                    	
 	Container Security:       	true                     	
 	Container Default Posture:	block(File)              	block(Capabilities)	block(Network)	
 	Host Default Posture:     	audit(File)              	audit(Capabilities)	audit(Network)	
 	Host Visibility:          	none                     	
Armored Up pods : 
+-------------+--------------------------------+------------+-----------------------------------------------------+--------+
|  NAMESPACE  |        DEFAULT POSTURE         | VISIBILITY |                        NAME                         | POLICY |
+-------------+--------------------------------+------------+-----------------------------------------------------+--------+
| kube-system | File(block),                   | none       | azure-wi-webhook-controller-manager-8dc4f68cf-2dnw7 |        |
|             | Capabilities(block), Network   |            |                                                     |        |
|             | (block)                        |            |                                                     |        |
+             +                                +            +-----------------------------------------------------+--------+
|             |                                |            | azure-wi-webhook-controller-manager-8dc4f68cf-dh7qn |        |
|             |                                |            |                                                     |        |
|             |                                |            |                                                     |        |
+             +                                +            +-----------------------------------------------------+--------+
|             |                                |            | csi-azuredisk-node-bckml                            |        |
|             |                                |            |                                                     |        |
|             |                                |            |                                                     |        |
+             +                                +            +-----------------------------------------------------+--------+
|             |                                |            | csi-azuredisk-node-bpfkw                            |        |
|             |                                |            |                                                     |        |
|             |                                |            |                                                     |        |
+             +                                +            +-----------------------------------------------------+--------+
|             |                                |            | csi-azurefile-node-h852w                            |        |
|             |                                |            |                                                     |        |
|             |                                |            |                                                     |        |
+             +                                +            +-----------------------------------------------------+--------+
|             |                                |            | csi-azurefile-node-rkjff                            |        |
|             |                                |            |                                                     |        |
|             |                                |            |                                                     |        |
+             +                                +            +-----------------------------------------------------+--------+
|             |                                |            | eraser-controller-manager-7d5c89b7f4-6mwgd          |        |
|             |                                |            |                                                     |        |
|             |                                |            |                                                     |        |
+             +                                +            +-----------------------------------------------------+--------+
|             |                                |            | konnectivity-agent-6d74b8f75c-hn9pd                 |        |
|             |                                |            |                                                     |        |
|             |                                |            |                                                     |        |
+             +                                +            +-----------------------------------------------------+--------+
|             |                                |            | konnectivity-agent-6d74b8f75c-m8s2h                 |        |
|             |                                |            |                                                     |        |
|             |                                |            |                                                     |        |
+             +                                +            +-----------------------------------------------------+--------+
|             |                                |            | konnectivity-agent-autoscaler-95f4794f-pxpzr        |        |
|             |                                |            |                                                     |        |
|             |                                |            |                                                     |        |
+-------------+--------------------------------+------------+-----------------------------------------------------+--------+
```

</details>

<details>
<summary>Test results</summary>

`blockposture`, `configmap`, `multicontainer`, `privileged`, `visibility`

```
Ginkgo ran 5 suites in 13m14.195067832s
Test Suite Passed
```

`throttling`

```
Ginkgo ran 1 suite in 54.128242374s
Test Suite Passed
```

`smoke`

```
Ginkgo ran 1 suite in 3m22.752403592s
Test Suite Passed
```

`syscalls`

```
Ran 17 of 19 Specs in 311.742 seconds
SUCCESS! -- 17 Passed | 0 Failed | 1 Flaked | 0 Pending | 2 Skipped
PASS

Ginkgo ran 1 suite in 5m15.580760094s
Test Suite Passed
```

`csp`

```
Ran 6 of 6 Specs in 244.303 seconds
SUCCESS! -- 6 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS

Ginkgo ran 1 suite in 4m8.178345598s
Test Suite Passed
```

`ksp` (failing tests manually verified)

```
Ran 30 of 42 Specs in 1633.220 seconds
FAIL! -- 27 Passed | 3 Failed | 8 Flaked | 0 Pending | 12 Skipped
--- FAIL: TestKsp (1633.22s)
FAIL

Ginkgo ran 1 suite in 27m15.425113478s

Test Suite Failed
```

</details>

--- a/KubeArmor/BPF/system_monitor.c
+++ b/KubeArmor/BPF/system_monitor.c
@@ -1025,8 +1025,9 @@ static __always_inline int save_all_hashes_to_the_buffer(bufs_t *bufs_p, bool ad
     if (num_of_hashes_idx >= MAX_BUFFER_SIZE || num_of_hashes_idx + num_of_hashes >= MAX_BUFFER_SIZE)
         return -1;
 
-    u32 idx = num_of_hashes_idx & (MAX_BUFFER_SIZE - 1);
+    u32 idx = num_of_hashes_idx;
     volatile_reg(idx);
+    idx &= (MAX_BUFFER_SIZE - 1);
 
     if (bpf_probe_read(&(bufs_p->buf[idx]), sizeof(u8), &num_of_hashes) != 0)
     {

